### DISCUSSION TITLE
Error when enrolling vehicle

### DISCUSSION THREAD

**Discussion by danielclau** [llamafilm/tesla-http-proxy-addon#32] [Q&A] [answered]
Succesfully did all the precursor steps (setting up new duckdns domain and NGINX. Obtained Client ID and secret from Tesla Developer. Put these details into Tesla HTTP  Proxy config)

However,
Clicked "Generate OAuth token" on my mobile Home Assistant Web UI (Android)
Signed into my Tesla account in Chrome successfully and I got a refresh token in the log
But when I click "Enrol public key in your vehicle" it takes me to my Tesla app and gives me the error "Unable to Share Vehicle"

In Tesla HTTP Proxy logs, there is this error after Generating OAuth Token

"[00:58:31] werkzeug:INFO: 172.30.33.5 - - [14/Mar/2024 00:58:31] "GET /callback?locale=en-US&code=NA_988652ba28260b0a21b30927bac59f2c845f4f1235c643d486d86614642c&state=d64a3162316f4f77a77e93f3c25375a4&issuer=https://auth.tesla.com/oauth2/v3 HTTP/1.0" 200 -
[00:58:31] run:ERROR: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again."

Pointing browser to https://{FQDN}/.well-known/appspecific/com.tesla.3p.public-key.pem successfully returns my public key.
Anyone seen this error before or know of a solution?

![Screenshot_20240314_004143_Tesla](https://github.com/llamafilm/tesla-http-proxy-addon/assets/22875562/825b7ce4-2215-49ed-b660-09ae7e070896)

**Comment by llamafilm** [comment, marked as answer]
I have not seen that issue before.

One thing to confirm, have you tested the public key URL externally? I mean, if it’s not available externally, and you hit that URL from inside your LAN, it might still resolve correctly due to hairpin routing. 

That log output is strange. It shows a 200 successful response to the callback, followed immediately by 404 at the same timestamp. What do you see in your browser after doing the Oauth login? To rerun it now, switch on “regenerate auth” and restart the addon.

**Reply by danielclau** [reply]
Restarting the add-on and restarting the "Generate OAuth" step seems to have fixed it.
Thank you.

**Reply by iamed18** [reply]
+1 for me on just regenerating the oauth tokens and things became suddenly fine for the same stated issue.
